agent/consul/state/memdb.go

```diff
@@ -147,11 +145,33 @@ func (t topic) String() string {
 	return string(t)
 }
 
+var (
+	TopicServiceHealth topic = "topic-service-health"
+	// TODO: why is connect a separate topic?
+	TopicServiceHealthConnect topic = "topic-service-health-connect"
```

Comment:
I'd like to answer this one in the docstring for this topic, but I don't really know the answer. I assume it has to do with how it is consumed by the client.

Reply:
Yeah, directly this is equivalent to `/health/service/:service` vs `/health/connect/:service`. The first returns all the registered instances of the service directly, the second returns only "connect capable" instances for a service. That means either the side-car proxies that are deployed with the service instance or a connect-native app instnce directly.

One is for non-connect clients to discover instances, the other is for connect client to discover how to connect using mTLS with AuthZ (which might be via a proxy not the actual service registration directly).